x-pack/plugins/security/public/management/users/components/confirm_delete_users/confirm_delete_users.tsx

```diff
@@ -81,27 +81,30 @@ export class ConfirmDeleteUsers extends Component<Props, unknown> {
   private deleteUsers = () => {
     const { usersToDelete, callback, userAPIClient, notifications } = this.props;
     const errors: string[] = [];
-    usersToDelete.forEach(async (username) => {
-      try {
-        await userAPIClient.deleteUser(username);
-        notifications.toasts.addSuccess(
-          i18n.translate(
-            'xpack.security.management.users.confirmDelete.userSuccessfullyDeletedNotificationMessage',
-            { defaultMessage: 'Deleted user {username}', values: { username } }
-          )
-        );
-      } catch (e) {
-        errors.push(username);
-        notifications.toasts.addDanger(
-          i18n.translate(
-            'xpack.security.management.users.confirmDelete.userDeletingErrorNotificationMessage',
-            { defaultMessage: 'Error deleting user {username}', values: { username } }
-          )
-        );
-      }
-      if (callback) {
-        callback(usersToDelete, errors);
-      }
-    });
+    Promise.all(
+      usersToDelete.map(async (username) => {
+        try {
+          await userAPIClient.deleteUser(username);
+          notifications.toasts.addSuccess(
+            i18n.translate(
+              'xpack.security.management.users.confirmDelete.userSuccessfullyDeletedNotificationMessage',
+              { defaultMessage: 'Deleted user {username}', values: { username } }
+            )
+          );
+        } catch (e) {
+          errors.push(username);
+          notifications.toasts.addDanger(
+            i18n.translate(
+              'xpack.security.management.users.confirmDelete.userDeletingErrorNotificationMessage',
+              { defaultMessage: 'Error deleting user {username}', values: { username } }
+            )
+          );
+        }
+
+        if (callback) {
+          callback(usersToDelete, errors);
+        }
```

Comment:
Noticed that this callback is being executed for every individual delete, any idea if that's intentional or should this wait for all deletions to succeed or fail before calling the callback?